### PR TITLE
Change `PatientsTabScreen` to use Mobius view effects

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/4653/change-patientstabscreen-to-use-mobius-view-effects

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/home/patients/PatientsEffectHandler.kt [hunk 1]
```diff
@@ -58,7 +58,7 @@ class PatientsEffectHandler @AssistedInject constructor(
         .addTransformer(LoadNumberOfPatientsRegistered::class.java, loadNumberOfPatientsRegistered())
         .addTransformer(LoadInfoForShowingAppUpdateMessage::class.java, loadInfoForShowingAppUpdate())
         .addConsumer(TouchAppUpdateShownAtTime::class.java, { appUpdateDialogShownAtPref.set(Instant.now(utcClock)) }, schedulers.io())
-        .addConsumer(PatientsTabViewEffect::class.java, viewEffectHandler::handle, schedulers.ui())
+        .addConsumer(PatientsTabViewEffect::class.java, viewEffectsConsumer::accept, schedulers.ui())
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/home/patients/PatientsEffectHandler.kt:59]: How would this work? Would accept function be internally calling handle?

**Review by Honey14 — COMMENTED** [review]
Looks good overall. Left one question, PTAL

File: app/src/main/java/org/simple/clinic/home/patients/PatientsEffectHandler.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsEffectHandler.kt:59]: Mobius will take care of that, if you look at the `BaseScreen`, we are observing the view effects live queue and dispatching them to our view effect handler. The live queue we are observing is a lifecycle aware component.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsEffectHandler.kt:59]: You can refer this to learn more: https://github.com/spotify/mobius/pull/78

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**